### DISCUSSION TITLE
[my mistake] WebApplicationBuilder.Build exits application with error code 1

### DISCUSSION THREAD

**Discussion by pvasek** [dotnet/aspnetcore#53522] [Q&A]
It seems that `WebApplicationBuilder.Build` does a lot of work, unfortunately, there is almost no logging. It's like a huge black box.

We encountered several times the issue that `Build` method failed but returned no exception.

I have no doubt it's an issue in our code, I just have no idea how to track it down. Any ideas, on how to investigate the `WebApplicationBuilder.Build` method?  

## The detailed description of the issue:

Right now we have this issue in our docker image in which it just exits the process with error code 1. No message is logged.

I added the following code, thinking the last error was not logged or something similar.

```C#
            Log.Information("Building application...");
            builder.Logging.ClearProviders();
            builder.Logging.AddConsole();
            builder.Logging.SetMinimumLevel(LogLevel.Trace);
            WebApplication app;
            try
            {
                app = builder.Build();
            }
            catch (Exception ex)
            {
                Log.Information("Building application failed");
                Log.Fatal(ex, "Building application failed");
                return 2;
            }
            Log.Information("Building application - Done");
```

But still, I can see only:
```
web-1  | [10:09:50 INF] Building application...
web-1 exited with code 1
```

**Comment by martincostello** [comment]
Try moving these lines into the `try` block in case they are where the exception is coming from?

```
builder.Logging.ClearProviders();
builder.Logging.AddConsole();
builder.Logging.SetMinimumLevel(LogLevel.Trace);
```

**Reply by pvasek** [reply]
Sorry, as I expected it was an issue with our code. The problem was that logging somehow stopped working after calling `build.Build()` when I replaced with just `Console.WriteLine` I could see that the call finished. 

I found it thanks to the article from Andrew Lock about [A brief introduction to DiagnosticSource](https://andrewlock.net/a-brief-introduction-to-diagnostic-source/) just by comparing traces I have seen during debugging and in the container.
